providers/src/airflow/providers/amazon/aws/sensors/sagemaker_unified_studio.py

```diff
@@ -0,0 +1,67 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""This module contains the Amazon SageMaker Unified Studio Notebook sensor."""
+
+from airflow import AirflowException
+from airflow.providers.amazon.aws.hooks.sagemaker_unified_studio import (
+    SageMakerNotebookHook,
+)
+from airflow.sensors.base import BaseSensorOperator
+from airflow.utils.context import Context
+
+
+class SageMakerNotebookSensor(BaseSensorOperator):
```

Comment:
Use `AwsBaseSensor`?

Reply:
Do we need the `AwsBaseSensor` for this one though? We're not making any AWS specific calls, so there's no need for specific AWS connection management. Only item we need is the `poke()` method (everything else is abstracted out to the Hook)